apps/cowswap-frontend/src/modules/bridge/components/BridgeRouteBreakdown.tsx

```diff
@@ -0,0 +1,444 @@
+  isAddress,
+  shortenAddress,
+  tryParseCurrencyAmount,
+} from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { TokenLogo } from '@cowprotocol/tokens'
+import { FiatAmount, InfoTooltip, UI } from '@cowprotocol/ui'
+
+import { ConfirmDetailsItem } from 'modules/trade/pure/ConfirmDetailsItem'
+import { ReceiveAmountTitle } from 'modules/trade/pure/ReceiveAmountTitle'
+import { DividerHorizontal } from 'modules/trade/pure/Row/styled'
+import { useUsdAmount } from 'modules/usdAmount'
+
+import { ProtocolIcons } from 'common/pure/ProtocolIcons'
+
+import {
+  AmountWithTokenIcon,
+  ArrowIcon,
+  Link,
+  NetworkLogoWrapper,
+  RecipientWrapper,
+  RouteHeader,
+  RouteTitle,
+  StopTitle,
+  StopsInfo,
+  TokenFlowContainer,
+  Wrapper,
+} from './styled'
+
+import { BridgeFeeType, BridgeProtocolConfig } from '../types'
+
+/**
+ * Check if a fee is free (zero)
+ * @param fee The fee value or fee type
+ * @returns True if the fee is free
+ */
+export function isFreeSwapFee(fee: string | BridgeFeeType): boolean {
+  return fee === BridgeFeeType.FREE || fee === 'FREE' || fee === '0' || fee === '0.0'
+}
+
+/**
+ * Get the CSS color variable for a fee based on its type
+ * @param fee The fee value or fee type
+ * @returns The CSS color variable or undefined if no special color applies
+ */
+export function getFeeTextColor(fee: string | BridgeFeeType): string | undefined {
+  if (isFreeSwapFee(fee)) {
+    return `var(${UI.COLOR_GREEN})`
+  }
+
+  return undefined
+}
+
+/**
+ * NetworkLogo component displays a network logo
+ */
+function NetworkLogo({ chainId, size = 16 }: { chainId: SupportedChainId; size?: number }) {
+  const theme = useTheme()
+  const chainInfo = getChainInfo(chainId)
+  const logoUrl = theme.darkMode ? chainInfo.logo.dark : chainInfo.logo.light
+
+  return (
+    <NetworkLogoWrapper size={size}>
+      <img src={logoUrl} alt={`${chainInfo.label} network logo`} />
+    </NetworkLogoWrapper>
+  )
+}
+
+/**
+ * Props for the BridgeRouteBreakdown component
+ */
+export interface BridgeRouteBreakdownProps {
+  // Swap details
+  sellAmount: string
+  sellToken: string
+  sellTokenAddress: string
+  buyAmount: string
+  buyToken: string
+  buyTokenAddress: string
+  networkCost: string
+  networkCostUsd: string
+  swapMinReceive?: string
+  swapExpectedToReceive?: string
+  swapMaxSlippage?: string
+
+  // Bridge details
+  bridgeAmount: string
+  bridgeToken: string
+  bridgeTokenAddress: string
+  bridgeReceiveAmount: string
+  bridgeTokenReceiveAddress?: string
+  bridgeFee: string | BridgeFeeType
+  maxBridgeSlippage: string
+  estimatedTime: number
+  recipient: string
+  bridgeProvider: BridgeProtocolConfig
+
+  // Optional props with defaults
+  recipientChainId?: SupportedChainId
+  sourceChainId?: SupportedChainId
+  tokenLogoSize?: number
+}
+
+/**
+ * BridgeRouteBreakdown component displays the details of a bridge transaction
+ */
+export function BridgeRouteBreakdown({
```

Comment:
The component needs to be split into smaller components, for example: BridgeSwapDetails and BridgeDetails

Reply:
Addressed in https://github.com/cowprotocol/cowswap/pull/5627